File: 1. Two Sum/java/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int newtarget = target - nums[i];
            if(numMap.containsKey(newtarget)){
                return new int[] {numMap.get(newtarget), i};
            }
            numMap.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: 1. Two Sum/cpp/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        for (int i = 0; i < nums.size(); i++) {
            int newtarget = target - nums[i];
            if (numMap.count(newtarget)) return {numMap[newtarget], i};
            numMap[nums[i]] = i;
        }
        return {};
    }
};